src/main/java/CreationShip/demo/NIO/Server.java

```diff
@@ -0,0 +1,156 @@
+package CreationShip.demo.NIO;
+
+import CreationShip.demo.NIO.comunic.Reader;
+import CreationShip.demo.NIO.comunic.Writer;
+import CreationShip.demo.NIO.worcker.Stages.AskQuestionConnector;
+import CreationShip.demo.NIO.worcker.Connection;
+import CreationShip.demo.service.MessageService;
+import CreationShip.demo.service.QuestionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.*;
+
+@Service
+public class Server extends Thread {
+
+    private static ThreadLocal<Integer> countConnection = new ThreadLocal<>();
+    private static InetSocketAddress inetSocketAddress = new InetSocketAddress("localhost", 8080);
+
+    @Autowired
+    private MessageService messageService;
+
+    @Autowired
+    private QuestionService questionService;
+
+
+    private ServerSocketChannel serverSocket = ServerSocketChannel.open();
+    private int allocate;
+
+    public Server(int allocate, InetSocketAddress inetSocketAddress) throws IOException {
+
+        this.allocate = allocate;
+        if (inetSocketAddress == null)
+            inetSocketAddress = new InetSocketAddress("localhost", 8080);
+        serverSocket.bind(inetSocketAddress);
+        serverSocket.configureBlocking(false);
+
+    }
+
+    public Server() throws IOException {
+
+        this.allocate = 64;
+
+        serverSocket.bind(inetSocketAddress);
+        serverSocket.configureBlocking(false);
+
+
+    }
+
+    public static Integer getCountConnection() {
+        return countConnection.get();
+    }
+
+    public void setAllocate(int allocate) {
+        this.allocate = allocate;
+    }
+
+    @Override
+    public void run() {
+
+        Selector selector;
+        ByteBuffer buffer;
+        countConnection.set(0);
+        Map<SelectionKey, String> lastMessage = new HashMap<>();
+        Map<SelectionKey, AskQuestionConnector> connectorMap = new HashMap<>();
+
+        Map<SelectionKey,Connection> connectionMap = new HashMap<>();
+
+        try {
+            selector = Selector.open();
+            serverSocket.register(selector, SelectionKey.OP_ACCEPT);
+            buffer = ByteBuffer.allocate(allocate);
+        } catch (IOException e) {
+            System.out.println(e);
+            return;
+        }
+
+        try {
+
+            while (true) {
+
+
+                selector.select();
+                Set<SelectionKey> selectedKeys = selector.selectedKeys();
+                Iterator<SelectionKey> iter = selectedKeys.iterator();
+
+                while (iter.hasNext()) {
+
+                    Reader reader = null;
+                    Writer writer = null;
+
+                    SelectionKey key = iter.next();
+
+
```

Comment:
Remove empty lines please. Also remove them in every class that has 3 or more empty lines in a row